src/st/cat/unk_3872C.c

```diff
@@ -1,35 +1,56 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 #include "cat.h"
 
-INCLUDE_ASM("st/cat/nonmatchings/unk_3872C", func_us_801B872C);
+extern EInit g_EInitSpawner;
+
+// This entity does not appear in the entity list and looks to be unused
+// It is not present in the PSP version.
+void func_us_801B872C(Entity* self) {
+    Entity* newEntity;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitParticle);
+        break;
+    case 1:
+        if (!(Random() & 7)) {
+            newEntity = AllocEntity(&g_Entities[STAGE_ENTITY_START],
+                                    &g_Entities[TOTAL_ENTITY_COUNT]);
+            if (newEntity != NULL) {
+                CreateEntityFromEntity(E_EXPLOSION, self, newEntity);
+                newEntity->params = (Random() & 3) - 0x4000;
+            }
+        }
+        break;
+    }
+}
 
 void func_us_801B87E8(Entity* self) {
```

Comment:
Brought in PSP fixes from @PancakeFriday's scratch https://decomp.me/scratch/Zbl5A